### ISSUE TITLE
Index out of range

### ISSUE THREAD

**Issue by banditopazzo** [Reeywhaar/nut#4]
Hi, I got this error:

```console
thread 'tokio-runtime-worker' panicked at /home/user1/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nut-0.1.1/src/db/db.rs:731:72:
range start index 57344 out of range for slice of length 32768
```
The error points to this line:  https://github.com/Reeywhaar/nut/blob/0dac83cfc8446fa9e99e72deeb4a1e678ddcb106/src/db/db.rs#L731

**Comment by banditopazzo** [comment]
The database is not corrupted. It works with the latest `bbolt`

**Comment by reeywhaar** [comment]
Well, I was making fork of [boltdb](https://github.com/boltdb/bolt) (now archived) at a time which is now a predecessor of bbolt. Can you confirm that same db works fine with original bolt?

**Comment by banditopazzo** [comment]
I made some tests.

- `nut`:
```console
❯ nut pages --path failing_bolt_state.db
ID     Type        Count   Overflow
------ ----------- ------- --------
     0 Flags(META)       0        0
     1 Flags(META)       0        0
     2 Flags(LEAVES)       1        0
     3 Flags(LEAVES)      13        0
     4 Flags(FREELIST)      15        0
     5 Flags(FREELIST)      15        0
     6 Flags(FREELIST)   24933 1918854516
     7 Flags(FREELIST)   24400 1599357761
Segmentation fault (core dumped)
```

- `bolt`:
```console
❯ bolt pages failing_bolt_state.db
ID       TYPE       ITEMS  OVRFLW
======== ========== ====== ======
0        meta       0            
1        meta       0            
2        leaf       1            
3        leaf       13           
4        free                    
5        freelist   15           
6        free                    
7        free                    
8        free                    
9        free                    
10       leaf       4      3     
14       free                    
15       free                    
16       free                    
17       free                    
18       free                    
19       free                    
20       free                    
21       free                    
22       free                    
23       free
```

- `bbolt`:
```console
❯ bbolt pages failing_bolt_state.db
ID       TYPE       ITEMS  OVRFLW
======== ========== ====== ======
0        meta       0            
1        meta       0            
2        leaf       1            
3        leaf       13           
4        freelist   15           
5        free                    
6        free                    
7        free                    
8        free                    
9        free                    
10       leaf       4      3     
14       free                    
15       free                    
16       free                    
17       free                    
18       free                    
19       free                    
20       free                    
21       free                    
22       free                    
23       free
```

As you can see `nut` is got a segfault, while `bolt` and `bbolt` were able to read the database. 
Please note also the different behaviour in the output of `bolt` and `bbolt`

**Comment by reeywhaar** [comment]
Well, freelist 24933 is definitely where the problem is. Care to share the db? if it doesn't have any private data of course.

**Comment by banditopazzo** [comment]
It's an almost empty `libpod` state. Here is it:
[failing_bolt_state.db.zip](https://github.com/Reeywhaar/nut/files/14068849/failing_bolt_state.db.zip)

**Comment by reeywhaar** [comment]
Fixed it. It was a copy paste error.

**Comment by banditopazzo** [comment]
Thank you @Reeywhaar !! 

PS: is this output normal?
```console
❯ nut pages --path failing_bolt_state.db
ID     Type        Count   Overflow
------ ----------- ------- --------
     0 Flags(META)       0        0
     1 Flags(META)       0        0
     2 Flags(LEAVES)       1        0
     3 Flags(LEAVES)      13        0
     4 Flags(FREELIST)      15        0
     5 Flags(FREELIST)      15        0
     6 Flags(FREELIST)   24933 1918854516
     7 Flags(FREELIST)   24400 1599357761
     8 Flags(FREELIST)   14383 842098225
     9 Flags(FREELIST)       2        0
    10 Flags(LEAVES)       4        3
    11 Flags(LEAVES | FREELIST | 0x2220)    8748 1952802668
    12 Flags(BRANCHES | LEAVES | 0x7460)   28521 574235246
    13 Flags(LEAVES | META | 0x6168)   25965 790768674
    14 Flags(FREELIST)       4        3
    15 Flags(FREELIST)   24933 1918854516
    16 Flags(FREELIST)   24400 1599357761
    17 Flags(FREELIST)   14383 842098225
    18 Flags(FREELIST)      13        0
    19 Flags(FREELIST)      11        0
    20 Flags(FREELIST)   24400 1599357761
    21 Flags(FREELIST)   12600 892481894
    22 Flags(FREELIST)   24400 1599357761
    23 Flags(FREELIST)   24948 577992052
```

**Comment by reeywhaar** [comment]
I guess I should add check if page was freed and then print it as “free” in `nut pages`

**Comment by reeywhaar** [comment]
Also something with `LEAVES | FREELIST | 0x2220` 🤔

**Comment by reeywhaar** [comment]
Fixed printing problem in 0.1.3

**Comment by banditopazzo** [comment]
Great! thanks
